pypowerwall/tedapi/vitals_dictionary.py

```diff
@@ -0,0 +1,517 @@
+            V_D = p['PVAC_Logging']['PVAC_PVMeasuredVoltage_D']
+            I_A = p['PVAC_Logging']['PVAC_PVCurrent_A']
+            I_B = p['PVAC_Logging']['PVAC_PVCurrent_B']
+            I_C = p['PVAC_Logging']['PVAC_PVCurrent_C']
+            I_D = p['PVAC_Logging']['PVAC_PVCurrent_D']
+            P_A = calculate_dc_power(V_A, I_A)
+            P_B = calculate_dc_power(V_B, I_B)
+            P_C = calculate_dc_power(V_C, I_C)
+            P_D = calculate_dc_power(V_D, I_D)
+            pvac[pvac_name] = {
+                "PVAC_Fout": lookup(p, ['PVAC_Status', 'PVAC_Fout']),
+                "PVAC_GridState": None,
+                "PVAC_InvState": None,
+                "PVAC_Iout": None,
+                "PVAC_LifetimeEnergyPV_Total": None,
+                "PVAC_PVCurrent_A": I_A,
+                "PVAC_PVCurrent_B": I_B,
+                "PVAC_PVCurrent_C": I_C,
+                "PVAC_PVCurrent_D": I_D,
+                "PVAC_PVMeasuredPower_A": P_A, # computed
+                "PVAC_PVMeasuredPower_B": P_B, # computed
+                "PVAC_PVMeasuredPower_C": P_C, # computed
+                "PVAC_PVMeasuredPower_D": P_D, # computed
+                "PVAC_PVMeasuredVoltage_A": V_A,
+                "PVAC_PVMeasuredVoltage_B": V_B,
+                "PVAC_PVMeasuredVoltage_C": V_C,
+                "PVAC_PVMeasuredVoltage_D": V_D,
+                "PVAC_Pout": lookup(p, ['PVAC_Status', 'PVAC_Pout']),
+                "PVAC_PvState_A": pvac_strings[pvac_name]["PVAC_PvState_A"], # These are computed from PVS below
+                "PVAC_PvState_B": pvac_strings[pvac_name]["PVAC_PvState_B"], #
+                "PVAC_PvState_C": pvac_strings[pvac_name]["PVAC_PvState_C"], # PV_Disabled, PV_Active, PV_Active_Parallel
+                "PVAC_PvState_D": pvac_strings[pvac_name]["PVAC_PvState_D"], # Not available in TEDAPI
+                "PVAC_Qout": None,
+                "PVAC_State": lookup(p, ['PVAC_Status', 'PVAC_State']),
+                "PVAC_VHvMinusChassisDC": None,
+                "PVAC_VL1Ground": lookup(p, ['PVAC_Logging', 'PVAC_VL1Ground']),
+                "PVAC_VL2Ground": lookup(p, ['PVAC_Logging', 'PVAC_VL2Ground']),
+                "PVAC_Vout": lookup(p, ['PVAC_Status', 'PVAC_Vout']),
+                "alerts": lookup(p, ['alerts', 'active']) or [],
+                "PVI-PowerStatusSetpoint": None,
+                "componentParentDin": None, # TODO: map to TETHC
+                "firmwareVersion": None,
+                "lastCommunicationTime": None,
+                "manufacturer": "TESLA",
+                "partNumber": packagePartNumber,
+                "serialNumber": packageSerialNumber,
+                "teslaEnergyEcuAttributes": {
+                    "ecuType": 296
+                }
+            }
+        return pvac
+
+    def get_pvs(self):
+        pvs, _ = self.__get_pvs_and_pvac_strings()
+        return pvs
+
+    def __get_pvac_strings(self):
```

Comment:
Rather than relying on magic knowledge of the tuple, return a plain old object that simply contains both?

Reply:
Easy enough!